docs/reference/analysis/tokenfilters/synonym-graph-tokenfilter.asciidoc

```diff
@@ -19,37 +19,57 @@ only. If you want to apply synonyms during indexing please use the
 standard <<analysis-synonym-tokenfilter,synonym token filter>>.
 ===============================
 
-Synonyms are configured using a configuration file.
-Here is an example:
+[discrete]
+[[analysis-synonym-graph-define-synonyms]]
+==== Define synonyms sets
 
-[source,console]
---------------------------------------------------
-PUT /test_index
-{
-  "settings": {
-    "index": {
-      "analysis": {
-        "analyzer": {
-          "search_synonyms": {
-            "tokenizer": "whitespace",
-            "filter": [ "graph_synonyms" ]
-          }
-        },
-        "filter": {
-          "graph_synonyms": {
-            "type": "synonym_graph",
-            "synonyms_path": "analysis/synonym.txt"
-          }
-        }
-      }
+include::synonyms-format.asciidoc[]
+
+[discrete]
+[[analysis-synonym-graph-configure-sets]]
+==== Configure synonyms sets
+
+Synonyms can be configured using the <<synonyms-store-synonyms-api,synonyms API>>, a <<synonyms-store-synonyms-file,synonyms file>>, or directly <<synonyms-store-synonyms-inline,inlined>> in the token filter configuration.
+See <<synonyms-store-synonyms,store your synonyms set>> for more details on each option.
+
+Use `synonyms_set` configuration option to provide a synonym set created via Synonyms Management APIs:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym",
+      "synonyms_set": "my-synonym-set"
```

Comment:
We also need to add "updateable: true", otherwise a user will get an error.

Reply:
Also, good to add a note that synonyms_set option must be used with updateable: true